File: services/reservation/src/main/java/com/example/iter/reservation/domain/repository/ReceiptRepository.java
package com.example.iter.reservation.domain.repository;

import com.example.iter.reservation.domain.entity.Receipt;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ReceiptRepository extends JpaRepository<Receipt, Long> {
    Optional<Receipt> findByRentalId(Long rentalId);
}

File: services/reservation/src/main/kotlin/com/example/iter/reservation/domain/repository/ReceiptRepository.kt
package com.example.iter.reservation.domain.repository

import com.example.iter.reservation.domain.entity.Receipt
import org.springframework.data.jpa.repository.JpaRepository

import java.util.Optional

interface ReceiptRepository : JpaRepository<Receipt, Long> {
    fun findByRentalId(rentalId: Long): Optional<Receipt>
}
